### PR TITLE
Pass FizzServerContext using FizzServerQuicHandshakeContext

### PR DESCRIPTION
**Pull request by deadalnix**
This remove one more fizz specific element from the common API

Depends on #162

### CODE REVIEW

**Comment by xttjsn** [comment]
lgtm, @yangchi ?

**Comment by xttjsn** [comment]
@deadalnix Thanks for your contribution! We really appreciate it! Just a small suggestion for your next pull requests: if possible, could you maintain that each subsequent commits are on top of each other? e.g. I see #164 bases directly on #162 , but #165 also based directly on #162 . This has made it difficult for me to commit your PR because such structure makes the GH imports quite messy at our end. E.g. the `getReadRecordLayerEncryptionLevel()` method is present in #164 but not in #165 . Again, thank you for contributing to mvfst!

File: quic/fizz/server/handshake/FizzServerHandshake.cpp [hunk 1]
```diff
@@ -21,16 +23,16 @@ FizzServerHandshake::FizzServerHandshake(
...
 
 void FizzServerHandshake::initializeImpl(
-    std::shared_ptr<const fizz::server::FizzServerContext> context,
     HandshakeCallback* callback,
     std::unique_ptr<fizz::server::AppTokenValidator> validator) {
-  auto ctx = std::make_shared<fizz::server::FizzServerContext>(*context);
-  ctx->setFactory(cryptoFactory_.getFizzFactory());
-  ctx->setSupportedCiphers({{fizz::CipherSuite::TLS_AES_128_GCM_SHA256}});
-  ctx->setVersionFallbackEnabled(false);
+  auto context = std::make_shared<fizz::server::FizzServerContext>(
```
Comment (yangchi) [inline quic/fizz/server/handshake/FizzServerHandshake.cpp:28]: is there a reason why this cannot simply be `context_ = std::make_shared<blah>();`

File: quic/fizz/server/handshake/FizzServerQuicHandshakeContext.h [hunk 2]
```diff
@@ -20,6 +22,40 @@ class FizzServerQuicHandshakeContext
  public:
   std::unique_ptr<ServerHandshake> makeServerHandshake(
       QuicServerConnectionState* conn) override;
+
+  const std::shared_ptr<const fizz::server::FizzServerContext>& getContext()
```
Comment (yangchi) [inline quic/fizz/server/handshake/FizzServerQuicHandshakeContext.h:26]: Can this returns const ref? I think the only usage of the return value is to be used to copy contrusct another FizzServerContext.

**Review by yangchi — COMMENTED** [review]
2 nits

File: quic/fizz/server/handshake/FizzServerQuicHandshakeContext.h (hunk 2 above)
Comment (yangchi) [reply quic/fizz/server/handshake/FizzServerQuicHandshakeContext.h:26]: I realize you are following the ownership model of existing code. So the current version is fine as well. I'm just curious if you see a reason returning a const ref won't work.

File: quic/fizz/server/handshake/FizzServerHandshake.cpp (hunk 1 above)
Comment (deadalnix) [reply quic/fizz/server/handshake/FizzServerHandshake.cpp:28]: I remove the field later on in #167 so I'd have to revert back to a local variable anyway.

File: quic/fizz/server/handshake/FizzServerQuicHandshakeContext.h (hunk 2 above)
Comment (deadalnix) [reply quic/fizz/server/handshake/FizzServerQuicHandshakeContext.h:26]: It's only use for a test, so i think either is fine.

Comment (yangchi) [reply quic/fizz/server/handshake/FizzServerQuicHandshakeContext.h:26]: Well with this diff it is now used in FizzServerHandshake::initializeImpl as well. Anyway, like i said this was before this diff.

**Comment by yangchi** [comment]
accepted on phabricator

**Pull request closed without merging**